packages/driver/src/cy/multidomain/index.ts

```diff
@@ -109,9 +131,30 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
       // @ts-ignore
       Cypress.multiDomainCommunicator.on('command:update', updateCommand)
 
-      return new Bluebird((resolve) => {
+      return new Bluebird((resolve, reject) => {
         // @ts-ignore
-        Cypress.multiDomainCommunicator.once('run:domain:fn', resolve)
+        Cypress.multiDomainCommunicator.once('run:domain:fn', (err) => err ? reject(err) : resolve())
+
+        // @ts-ignore
+        Cypress.multiDomainCommunicator.once('unbind:done:called', () => {
+          // if done is to be called from the secondary domain, the 'done:called' event
+          // should have already been invoked in the switchToDomain function
+          // Go ahead and remove the listener
```

Comment:
I'm confused by this comment. Isn't this only used if done is called from the primary domain? If it's called from the secondary domain, it's automatically unbound since we use `.once`.

Reply:
Also, what happens in this case with the `deferredDone` promise? It won't ever resolve, right?

Reply:
Yeah... this comment doesn't really make a lot of sense especially on revisiting 😂 . And awesome catch! Looks like if `done` is provided and called in the primary domain, the `deferredDone` never settled. I am going to rework that comment to make sense with why that handle is being unbound, as well as make sure that promise is settled. I am thinking something like this:

```js
communicator.once('unbind:done:called', () => {
  // If `done()` is passed into the test, but is called in the primary domain and NOT the secondary domain
  // Go ahead and make sure the listener is removed
  communicator.off('done:called', invokeDone)

  // If done was NOT called in the secondary domain, go ahead and make sure this promise is settled.
  // if done was called in the secondary domain, this promise should already be settled and resolving it has no effect.
  deferredDone.resolve()
})
```

What do you think?